lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -290,31 +407,68 @@ def run(self, **kwargs):
                         if port:
                             self.update_pillar_and_conf('port', port)
                     else:
-                        self.logger.error(
-                            "Enclosure ID is not set: Please provide user and password as well to set Enclosure ID"
+                        self.logger.error("Enclosure ID is not set"
+                            "Run following command to set the enclosure id:"
+                            "cortx_setup storage config --user <user>"
+                            " --password <passwd> --ip <ip> --port <port>")
+                        raise RuntimeError(
+                            "Cannot set mode, ip and port without enclosure id"
                         )
-                        raise RuntimeError("Incomplete arguments provided")
 
                 if user is not None and password is not None:
                     if self.enclosure_id:
-                        host = pillar_get(f"storage/{enc_num}/controller/primary/ip")[node_id][f'storage/{enc_num}/controller/primary/ip']
-                        port = pillar_get(f"storage/{enc_num}/controller/primary/port")[node_id][f'storage/{enc_num}/controller/primary/port']
-
-                        valid_connection_check = EnclosureInfo(host, user, password, port).connection_status()
+                        host = get_pillar_data(
+                            f"storage/{enc_num}/controller/primary/ip"
+                            )[node_id][f'storage/{enc_num}/controller/primary/ip']
+                        port = get_pillar_data(
+                            f"storage/{enc_num}/controller/primary/port"
+                            )[node_id][f'storage/{enc_num}/controller/primary/port']
```

Comment:
get_pillar_data method requires just key of the pillar as an argument and returns its value.

so it should be like this:
```
host = get_pillar_data(f"storage/{enc_num}/controller/primary/ip")
```

instead of
```
host = get_pillar_data(
                      f"storage/{enc_num}/controller/primary/ip"
            )[node_id][f'storage/{enc_num}/controller/primary/ip']
```
get_pillar_data retuns value and not dictionary

Reply:
Thanks, updated.